### PR TITLE
Added first draft of ILDCP

### PR DESCRIPTION
**Pull request by dora-gt**
I wrote the first draft of RFC of ILDCP based on this issue: #431 , reading the source code.
Please check the description is correct or not.

### CODE REVIEW

**Comment by adrianhopebailie** [comment]
@dora-gt this is great, thanks!

I will make a quick pass of edits rather than just log lots of issues then let's get a review from @sharafian or @justmoon before we merge.

**Comment by dora-gt** [comment]
@adrianhopebailie thank you so much for your confirmation and edit!

File: 0031-dynamic-configuration-protocol/0031-dynamic-configuration-protocol.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+- A **node** is a participant in an Interledger network. It may be a [connector](../0001-interledger-architecture/0001-interledger-architecture.md#connectors), a sender or a receiver. 
+- Nodes are connected to each other and the relationship between two nodes is either:
+  - `parent` and `child` (one node is the `parent` of another node that is relatively its `child`), or,
+  - `peer` and `peer` (two nodes are peered with one another).
+
+## Overview
+Interledger is a protocol suite that consists of several protocols including the [Bilateral Transfer Protocol](../0023-bilateral-transfer-protocol/0023-bilateral-transfer-protocol.md), the Route Broadcasting Protocol and others necessary for establishment, operation and maintenance of the network. Interledger Dynamic Configuration Protocol (ILDCP) is one of these protocols.
+
+In order to participate in the network a node must have an ILP address. This address is part of a heirarchical address space where child nodes MAY be allocated addresses within the address space of their parent node. This makes routing on the network more efficient than if all nodes had top-level addresses.
```
Comment (sharafian) [inline 0031-dynamic-configuration-protocol/0031-dynamic-configuration-protocol.md]: Can we link to the ILP address spec from here?

File: 0031-dynamic-configuration-protocol/0031-dynamic-configuration-protocol.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+A node is therefor either configured with an address which it broadcasts to its peers or, if the node is a child of a parent peer-node, the ILP address of the child node can be allocated to it by the parent. This is done using ILDCP.
```
Comment (sharafian) [inline 0031-dynamic-configuration-protocol/0031-dynamic-configuration-protocol.md]: therefor -> therefore

Comment (sharafian) [inline 0031-dynamic-configuration-protocol/0031-dynamic-configuration-protocol.md]: if the node is a child of a parent peer-node -> if the node has a parent in the network hierarchy

File: 0031-dynamic-configuration-protocol/0031-dynamic-configuration-protocol.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+In short, **ILDCP is a protocol used for transferring node and ledger information from a parent node to a child node**. The transferred information is:
```
Comment (sharafian) [inline 0031-dynamic-configuration-protocol/0031-dynamic-configuration-protocol.md]: Worth noting that the protocol is also extensible so new fields can be added

**Comment by dora-gt** [comment]
@sharafian Thank you so much for the review. I'll fix soon.

File: 0031-dynamic-configuration-protocol/0031-dynamic-configuration-protocol.md (hunk 3 above)
Comment (dora-gt) [reply 0031-dynamic-configuration-protocol/0031-dynamic-configuration-protocol.md]: Should I use `...` in the ASN.1 definition though I don't know how that is interpreted in OER exactly though.

File: 0031-dynamic-configuration-protocol/0031-dynamic-configuration-protocol.md (hunk 2 above)
Comment (adrianhopebailie) [reply 0031-dynamic-configuration-protocol/0031-dynamic-configuration-protocol.md]: I think @sharafian is correct. Will fix

**Pull request merged**